web_src/js/features/common-global.js

```diff
@@ -60,6 +60,57 @@ export function initGlobalButtonClickOnEnter() {
   });
 }
 
+async function fetchFormAction(e) {
+  if (!e.target.classList.contains('form-fetch-action')) return;
+
+  e.preventDefault();
+  const formEl = e.target;
+  if (formEl.classList.contains('loading')) return;
+
+  // TODO: fine tune the UI feedback
+  formEl.classList.add('loading');
+  e.submitter?.classList.add('loading');
+
+  const formMethod = formEl.getAttribute('method') || 'get';
+  const formActionUrl = formEl.getAttribute('action');
+  const formData = new FormData(formEl);
+  const [submitterName, submitterValue] = [e.submitter?.getAttribute('name'), e.submitter?.getAttribute('value')];
+  if (submitterName) {
+    formData.append(submitterName, submitterValue);
+  }
+
+  const req = {method: formMethod.toUpperCase(), headers: {'X-Csrf-Token': csrfToken}};
+  if (formMethod === 'GET') {
+    // TODO: append to the action URL
+  } else {
+    req.body = formData;
+  }
+
+  const onError = () => {
+    // TODO: show error to end users
+    formEl.classList.remove('loading');
+    e.submitter?.classList.remove('loading');
+  };
+
+  try {
+    const resp = await fetch(formActionUrl, req);
+    if (resp.status === 200) {
+      const {redirect} = await resp.json();
```

Comment:
```suggestion
    const res = await fetch(formActionUrl, req);
    if (res.ok) {
      const {redirect} = await res.json();
```

Reply:
If it is only a nit, I do not want to use `res`, which reads like `result`, I really don't like it and never used it.

If it is a must, please help to add a lint rule or write a guideline document, to avoid suggesting it again and again ....

Reply:
My order of preference is `res` > `response` > `resp`.

Reply:
My order of preference is `resp` > `response` , no `res` because I only use `res` for "results".

And all Golang backend code also uses `resp/Resp`.

Reply:
I guess at some point, I will at some point make a fetch wrapper anyways that returns `data`, which will make this discussion obsolete.

Reply:
> I guess at some point, I will at some point make a fetch wrapper anyways that returns `data`

Then that's `$.post` and `$.ajax`

Reply:
My usual fetch wrappers are method names, e.g.

```js
data = await POST(url, opts)
```;

Reply:
I am curious is `$.post` or `$.ajax` more preferred than `fetch` in gitea?

Reply:
There are soooo many (grep "*.go") ....

![image](https://github.com/go-gitea/gitea/assets/2114189/c41f131d-a024-4733-a8fe-a902c84f02a7)

Reply:
Yeah let's do it than. Consistency over personal preference then.